### PR TITLE
DOCSP-45313: Remove mongocxx::stdx namespace

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-45313
Staging - https://deploy-preview-83--docs-cpp.netlify.app/upgrade/#version-4.0

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -159,6 +159,12 @@ The v4.0 driver introduces the following breaking changes to the API:
   library equivalents. For context, see the :ref:`Build System Breaking Changes
   <version-4.0-build-system-breaking>` section.
 
+- Removes the the ``mongocxx::stdx`` namespace. Replace the following
```
Comment (mongoKart) [inline source/upgrade.txt:162]:
```suggestion
- Removes the ``mongocxx::stdx`` namespace. Replace the following
```

File: source/upgrade.txt [hunk 2]
```diff
@@ -169,6 +175,12 @@ The v4.0 driver introduces the following breaking changes to the ABI:
...
```
Comment (mongoKart) [inline source/upgrade.txt:178]:
```suggestion
- Removes the ``mongocxx::stdx`` namespace. Replace the following
```

**Review by mongoKart — APPROVED** [review]
lgtm w/ fixes

File: source/upgrade.txt [hunk 3]
```diff
@@ -169,6 +175,12 @@ The v4.0 driver introduces the following breaking changes to the ABI:
   library equivalents. For context, see the :ref:`Build System Breaking Changes
   <version-4.0-build-system-breaking>` section.
 
+- Removes the ``mongocxx::stdx`` namespace. Replace the following
+  types in your code:
+
+  - ``mongocxx::stdx::optional<T>``. Instead, use ``bsoncxx::stdx::optional<T>``.
+  - ``mongocxx::stdx::string_view``. Instead, use ``bsoncxx::stdx::string_view``.
+
```
Comment (eramongodb) [inline source/upgrade.txt:183]:
```suggestion
```
This one is not an ABI breaking change, as the actual types are unchanged (via type aliasing).

**Review by eramongodb — APPROVED** [review]
One suggestion to address; otherwise, LGTM.

**Pull request merged**